low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,369 @@
+        status_list = req_register.get(["%s>status" % req_id])
+        status = status_list[0] if status_list else None
+
+        if not status:
+            raise SupportBundleError(
+                errno.EINVAL, "Request ID '%s' not found." % req_id)
+        else:
+            # Set failed status to stale request ID
+            expiry_sec = sspl_const.SB_REQUEST_TIMEOUT
+            last_reboot = int(psutil.boot_time())
+            # Set request is expired if processing time exceeds
+            req_start_time = int(time.mktime(time.strptime(
+                req_register.get(["%s>time" % req_id])[0],
+                '%Y-%m-%d %H:%M:%S')))
+            current_time = int(time.time())
+            is_req_expired = (current_time - req_start_time) > expiry_sec
+            if is_req_expired or (last_reboot > req_start_time and
+                                  status is BundleRequestHandler.IN_PROGRESS):
+                # Set request state as failed
+                BundleRequestHandler._update_request_status(
+                    req_id, "Failed - request is expired.")
+            status = req_register.get(["%s>status" % req_id])[0]
+
+        return status
+
+    @staticmethod
+    def _update_request_status(req_id, status):
+        """Updates processed request information."""
+        req_info = req_register.get(["%s" % req_id])[0]
+        req_info.update({
+            "status": status,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        })
+        req_register.set(["%s" % req_id], [req_info])
+
+    @staticmethod
+    def _add_bundle_request(bundle_id, req_id, target_path):
+        """Updates new request information."""
+        req_info = {
+            "request_id": req_id,
+            "target_path": target_path,
+            "status": BundleRequestHandler.IN_PROGRESS,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        }
+        req_register.set(["%s" % bundle_id], [req_info])
+
+
+class SupportBundle(object):
+    """SSPL support bundle class"""
+
+    def __init__(self):
+
+        self.SYS_INFORMATION = "SYSTEM_INFORMATION"
+        self.IEM_SENSOR = "IEMSENSOR"
+        self.bundle_prefix = "sb_"
+        self._tmp_dir = '/tmp/cortx'
+        self._default_path = '%s/support_bundle/' % self._tmp_dir
+        self.sspl_conf_dir = f"{sspl_const.SSPL_SB_TMP}/sspl_conf/"
+        self.ipmi_sel_data = "{0}/ipmi_sel_data.txt".format(
+            sspl_const.SSPL_SB_TMP)
+        self.boot_drvs_dta = "{0}/Server_OS_boot_drives_SMART_data".format(
+            sspl_const.SSPL_SB_TMP)
+        self.enclosure_log = f"{sspl_const.SSPL_SB_TMP}/enclosure_logs.zip"
+        self.enclosure_wwn = "NA"
+        self.sspl_log_dir = Conf.get(SSPL_CONF, "%s>sspl_log_file_path" %
+                                     (self.SYS_INFORMATION)).replace("/sspl.log", "")
+        self.iem_log_dir = Conf.get(SSPL_CONF, "%s>log_file_path" %
```

Comment:
IEM collection should be optional now. Add TODO here to remove IEM log collection once SSPL IEM sensor is disabled

Reply:
Done